frontend/src/lib/cluster.ts

```diff
@@ -0,0 +1,30 @@
+import { matchPath } from 'react-router';
```

Comment:
What does this commit do?? How was there a circular dependency before?? And what was causing it??

Reply:
> frontend: lib: Add lib/cluster to avoid circular imports from utils

With the changes in this PR it avoids the circular dependency where something imports utils, and utils then imports that thing, then that thing imports utils again - ad infinitum.